pkg/instancetype/instancetype.go

```diff
@@ -1534,28 +1534,38 @@ func applyFirmwarePreferences(preferenceSpec *instancetypev1beta1.VirtualMachine
 		return
 	}
 
+	firmware := preferenceSpec.Firmware
+
 	if vmiSpec.Domain.Firmware == nil {
 		vmiSpec.Domain.Firmware = &virtv1.Firmware{}
 	}
 
-	if vmiSpec.Domain.Firmware.Bootloader == nil {
-		vmiSpec.Domain.Firmware.Bootloader = &virtv1.Bootloader{}
+	vmiFirmware := vmiSpec.Domain.Firmware
```

Comment:
Move to L1538 and use it in the block above too? Sorry, did not notice this earlier.

Reply:
Sorry for the noise, won't work if `vmiSpec.Domain.Firmware` is `nil`.